docs/operations.py

```diff
@@ -0,0 +1,180 @@
+            for k, v in config.items('filesystem'):
+                if v == 'True':
+                    self.fstype = k
+            if self.fstype == 'ceph':
+                self.rid = config.get(self.fstype, 'id')
+                self.pool = config.get(self.fstype, 'pool')
+                self.r_conf = config.get(self.fstype, 'conf_file')
+                self.keyring = config.get(self.fstype, 'keyring')
+        except ConfigParser.NoOptionError, err: #which is same as 'exp as e'
+            print str(err)
+
+#Calling shell script which executes a iscsi update as we don't have 
+#rbd map in documentation.
+def call_shellscript(path, m_args):
+        arglist = []
+        arglist.append(path)
+        for arg in m_args:
+                arglist.append(arg)
+        proc = subprocess.Popen(arglist, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        return proc.communicate()
+
+#Custom Exception Class that we are going to use which is a wrapper around
+#Ceph exception classes.
+class ResponseException(object):
+    def __init__(self, e):
+        self.exception_dict = dict()
+        self.exception = e
+# Extend this dict as needed for future expceptions. this
+# ensures readeability and uniformity of errors.
+        self.exception_dict = {
+           type(rbd.ImageExists())  : 401,
+           type(CephException()) : 403,
+           type(rbd.ImageBusy()) : 409,
+           type(rbd.ImageHasSnapshots()) : 405,
+           type(rbd.ImageNotFound()) : 404,
+           type(rbd.FunctionNotSupported()) : 410,
+           type(rbd.ArgumentOutOfRange()) : 411,
+           type(ShellScriptException()) : 440,
+                                }
+        #This is to handle key error exception, this also gives us the default error
+        self.current_err = self.exception_dict.get(type(e), 500)
+
+    def parse_curr_err(self, debug = False):
+        if self.current_err != 500:  #check for existing error code in the exception, 
+            emsg = self.exception.message
+        elif self.current_err == 500:
+            #This is for debugging original exceptions, if we don't have this, 
+            #we will always get internal error, so that end-user can't 
+            #see original exceptions
+            if debug == True:
+                print e
+            emsg = "Internal server error"
+        return {'status_code' : self.current_err, 'msg' : emsg}
+
+
+#Provisioning the nodes for a given project using the image and snapshot name
+#given. The nodes are typically implemented using ceph.
+def provision(node_name, img_name = "hadoopMaster.img",\
+        snap_name = "HadoopMasterGoldenImage",\
+        debug = False):
+    try:
+        fsconfig = create_fsconfigobj()
+        fs_obj = init_fs(fsconfig)
+        if fs_obj.clone(img_name.encode('utf-8'),\
+                snap_name.encode('utf-8'),\
+                node_name.encode("utf-8")):
+            iscsi_output = call_shellscript('/home/user/ims/iscsi_update.sh', \
```

Comment:
@genagain Valid point, thats something that we need to fix but this is more less going to be a VM that will be downloaded by user and this has prepackaged stuff, so the path may not change at all but we will fix that out.